mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/MantisMasterClientApi.java

```diff
@@ -699,16 +699,24 @@ public Observable<String> getJobStatusObservable(final String jobId) {
      * @return
      */
     public Observable<JobSchedulingInfo> schedulingChanges(final String jobId) {
+        final ConditionalRetry retryObject = new ConditionalRetry(null, "assignmentresults_" + jobId);
         return masterMonitor.getMasterObservable()
                 .filter(masterDescription -> masterDescription != null)
                 .retryWhen(retryLogic)
```

Comment:
There are multiple calls to `retryWhen`, do you need to change all of them?

I want to understand this scenario more. In a typical case, when a jobId is gone, it's because it is replaced by a new job (SLA 1/1). In that case, the old connection is properly closed, and we should unsubscribe to this observable normally. From your investigation, when are we fetching assignments indefinitely?

Reply:
(there isn't enough logs available to prove it so this is only my theory): the job IDs being queried seem to belong to the jobs that went through bad deployments (e.g. SourceJobA-3 to SourceJobA-4, where 4 was in a bad state and then the source job was updated to SourceJobA-5). In this case, the observable chain to SourceJobA-4 wasn't closed/unsub properly so the query part was kept alive in the upper switchMap and stuck in its infinite retry. 

This can explain why only a small subset of source job IDs are being constantly queried in control plane logs (the ones not getting closed correctly).
Thus I think i only need to handle the one retry policy responsible for this infinite retry (the added test was stuck before the change and now can close the observable correctly).